File: BasicClientServerProgram/Java/Server_thread.java
import java.io.*;
import java.util.Scanner;


//network libraries
import java.net.ServerSocket;
import java.net.Socket;

class Serverthread{

    public static void main(String args[]){

        Scanner sc = new Scanner(System.in);
    

        ServerSocket fd_server = null;


        try{
            fd_server = new ServerSocket(6379);
        }catch(IOException e1){
            System.out.println("Error in socket activation");
        }

    while(true){
    Socket temp = null;

      if(fd_server != null){
        try{
             temp = fd_server.accept();
             System.out.println("Client connected.....");
        }catch(IOException e2){
            System.out.println("Error during client connection");
        }      
      }

    final Socket fd_client = temp;
    Thread t = new Thread(()->{
        
        Msghandlerthread msg = new Msghandlerthread(fd_client);
        String m = null;

         boolean b = true;
         while(b){
                m = msg.receive_message(msg.r);
                if(m == null)
                b = false;
                else{
                System.out.println("Client "+Thread.currentThread().getId()+" says: "+m);
                msg.send_message(msg.w,"Message received");
                }
            }
        
        try{
            fd_client.close();
        }catch(IOException e5){
            System.out.println("Error closing a socket");
        }

    });
    t.start();
  }    
 }

}

File: BasicClientServerProgram/Python/Serverthread.py
import os
import sys
import socket
import threading

from inc.Msghandler import Msghandler

SERVER_ADDR = "IP_NUM"
PORT = PORT_NUM
msg = Msghandler(1024)
c_num = 0

def handle_client(soc,c_num):
    while True:
        m = msg.receive_message(fd_client)
        print("client ",c_num," says: ",m,end="\n")
        msg.send_message(fd_client,"Message received \n")
    
    soc.close()




try:
    s = socket.socket(socket.AF_INET,socket.SOCK_STREAM)
    s.bind((SERVER_ADDR,PORT))
    s.listen()

    while True:
        fd_client, client_addr = s.accept()
        print("A Client connected....\n")
        c_num  = c_num+1
        t = threading.Thread(target=handle_client,args=(fd_client,c_num))
        t.start()
    
    s.close()

except socket.error as e:
    print("Socket error: ",e,end="\n")
finally:
    s.close()

